packages/nimble-components/src/table-column/number-text/models/default-formatter.ts

```diff
@@ -0,0 +1,53 @@
+import { NumberFormatter } from './number-formatter';
+
+/**
+ * The formatter for a number-text column whose format is configured to be 'default'.
+ */
+export class DefaultFormatter extends NumberFormatter {
+    private static readonly maximumDigits = 6;
+    // Use exponential notation for numbers that will be displayed with
+    // 3 leading 0s or more because otherwise at least half of the displayed
+    // digits will be leading 0s.
+    private static readonly exponentialLowerBound = 0.000995;
+    // Use exponential formatting for numbers whose magnitude cannot
+    // otherwise be displayed with less than 6 digits or less.
+    private static readonly exponentialUpperBound = 999999.5;
+
+    private static readonly defaultFormatter = new Intl.NumberFormat(
+        undefined,
+        {
+            maximumSignificantDigits: DefaultFormatter.maximumDigits,
+            maximumFractionDigits: DefaultFormatter.maximumDigits - 1,
+            // Use 'lessPrecision' rounding priority to help ensure that no more than 6 digits are rendered
+            // in the formatted value, even if 'maximumSignificantDigits' or 'maximumFractionDigits' alone
+            // would result in more than 6 rendered digits.
+            roundingPriority: 'lessPrecision',
+            useGrouping: true,
+            // @ts-expect-error - The version of TypeScript currently being used does not include 'negative' as a valid signDisplay value
```

Comment:
Update: It looks like the value of `"negative"` isn't supported in the version of FF our tests run against (115), despite the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#browser_compatibility) claiming that it is. The value is, however, supported in the version of FF that is installed on my machine (116). I'm working on finding a good alternative.

Reply:
I ended up writing code to coerce `-0` to `0` before calling format since I couldn't use `signDisplay: 'negative'`.

I also realized that `roundingPriority` isn't supported in all browsers either, so I had to add an additional condition when using the `default` format for when to use 6 significant digits, 5 decimal digits, or exponential notation with 6 significant digits.